src/policy/rbf.cpp

```diff
@@ -42,3 +48,174 @@ RBFTransactionState IsRBFOptInEmptyMempool(const CTransaction& tx)
+        // above. See the comment above the first CalculateMempoolAncestors
+        // call for more info.
+        if (!setConflictsParents.count(tx.vin[j].prevout.hash))
+        {
+            // Rather than check the UTXO set - potentially expensive -
+            // it's cheaper to just check if the new input refers to a
+            // tx that's in the mempool.
+            if (m_pool.exists(tx.vin[j].prevout.hash)) {
+                return state.Invalid(TxValidationResult::TX_MEMPOOL_POLICY, "replacement-adds-unconfirmed",
+                        strprintf("replacement %s adds unconfirmed input, idx %d",
+                            tx.GetHash().ToString(), j));
+            }
+        }
+    }
+    return true;
+}
+
+bool SpendsAndConflictsDisjoint(CTxMemPool::setEntries& setAncestors, std::set<uint256> setConflicts,
+                                TxValidationState& state, const uint256& hash)
+{
+    for (CTxMemPool::txiter ancestorIt : setAncestors)
+    {
+        const uint256 &hashAncestor = ancestorIt->GetTx().GetHash();
+        if (setConflicts.count(hashAncestor))
+        {
+            return state.Invalid(TxValidationResult::TX_CONSENSUS, "bad-txns-spends-conflicting-tx",
+                    strprintf("%s spends conflicting transaction %s",
+                        hash.ToString(),
+                        hashAncestor.ToString()));
+        }
+    }
+    return true;
+}
+
+bool PaysMoreThanConflicts(const CTxMemPool::setEntries& setIterConflicting, CFeeRate newFeeRate,
+                           TxValidationState& state, const uint256& hash)
+{
+    for (const auto& mi : setIterConflicting) {
+        // Don't allow the replacement to reduce the feerate of the
+        // mempool.
+        //
+        // We usually don't want to accept replacements with lower
+        // feerates than what they replaced as that would lower the
+        // feerate of the next block. Requiring that the feerate always
+        // be increased is also an easy-to-reason about way to prevent
+        // DoS attacks via replacements.
+        //
+        // We only consider the feerates of transactions being directly
+        // replaced, not their indirect descendants. While that does
+        // mean high feerate children are ignored when deciding whether
+        // or not to replace, we do require the replacement to pay more
+        // overall fees too, mitigating most cases.
+        CFeeRate oldFeeRate(mi->GetModifiedFee(), mi->GetTxSize());
+        if (newFeeRate <= oldFeeRate)
+        {
+            return state.Invalid(TxValidationResult::TX_MEMPOOL_POLICY, "insufficient fee",
+                    strprintf("rejecting replacement %s; new feerate %s <= old feerate %s",
+                        hash.ToString(),
+                        newFeeRate.ToString(),
+                        oldFeeRate.ToString()));
+        }
+    }
+    return true;
+}
+
+bool PaysForRBF(CAmount nConflictingFees, size_t nConflictingSize,
```

Comment:
Note: `nConflictingSize` is an unused variable

Reply:
Removed - thanks!